src/convert.ts

```diff
@@ -41,12 +47,34 @@ export function convert(asyncapi: string | AsyncAPIDocument, version: ConvertVer
   fromVersion++;
   let converted = document;
   for (let i = fromVersion; i <= toVersion; i++) {
-    const v = conversionVersions[i] as ConvertVersion;
-    converted = converters[v](converted, options);
+    const v = conversionVersions[i] as AsyncAPIConvertVersion;
+    converted = asyncAPIconverters[v](converted, options);
   }
 
   if (format === 'yaml') {
     return dump(converted, { skipInvalid: true });
   }
   return converted;
 }
+
+export function convertOpenAPI(input: string ,version: OpenAPIConvertVersion,options?: OpenAPIToAsyncAPIOptions): string;
+export function convertOpenAPI(input: OpenAPIDocument, version: OpenAPIConvertVersion ,options?: OpenAPIToAsyncAPIOptions): AsyncAPIDocument;
+export function convertOpenAPI(input: string | OpenAPIDocument, version: OpenAPIConvertVersion, options: OpenAPIToAsyncAPIOptions = {}): string | AsyncAPIDocument {
+
+  const { format, document } = serializeInput(input);
+  const openApiVersion = document.openapi;
+  const converterVersion = openApiVersion === '3.0.0' ? '3.0.0' : openApiVersion;
+
+  const openapiToAsyncapiConverter = openapiConverters[converterVersion as OpenAPIConvertVersion] as ConvertOpenAPIFunction;
+
+  if (!openapiToAsyncapiConverter) {
+    throw new Error("OpenAPI to AsyncAPI converter is not available.");
```

Comment:
This is gonna be a user error, lets use something they understand not internal descriptions. Maybe something like:
`throw new Error(\`We are not able to convert OpenAPI ${converterVersion} to AsyncAPI, please raise a feature request.\`);`

Reply:
Done